cmd/kube-proxy/app/server.go

```diff
@@ -399,6 +399,15 @@ func NewProxyServer(config *componentconfig.KubeProxyConfiguration, cleanupAndEx
 	proxyMode := getProxyMode(string(config.Mode), iptInterface, iptables.LinuxKernelCompatTester{})
 	if proxyMode == proxyModeIPTables {
 		glog.V(0).Info("Using iptables Proxier.")
+		var nodeIP net.IP
+		if config.BindAddress != "0.0.0.0" {
+			nodeIP := net.ParseIP(config.BindAddress)
+			if local := isLocalIP(nodeIP); !local {
```

Comment:
I don't think we should bother to check this at all.  If the user specified an IP, use it.  Full stop.  If it is not a valid IP, it will fail (and you should test it) with a decent error.  That makes this PR pretty trivial.